Python/optimizer_analysis.c

```diff
@@ -301,6 +301,7 @@ sym_new_known_notnull(_Py_UOpsAbstractInterpContext *ctx)
     if (res == NULL) {
         return NULL;
     }
+    sym_set_flag(res, KNOWN);
     sym_set_flag(res, NOT_NULL);
```

Comment:
I know I'm late to this party, but in general most of the time we always assume that values on the stack are not NULL. IIRC there only exceptions are the NULLs inserted before call arguments. Maybe we should change the logic and nomenclature so that "may be NULL" is the special bit we manipulate and check?